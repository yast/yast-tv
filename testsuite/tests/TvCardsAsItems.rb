# encoding: utf-8

module Yast
  class TvCardsAsItemsClient < Client
    def main
      # testedfiles: Tv.ycp

      Yast.include self, "testsuite.rb"

      @READ = { "target" => { "size" => -1, "tmpdir" => "/tmp" } }

      TESTSUITE_INIT([@READ], nil)
      Yast.import "Tv"

      # Tv::cards == nil
      Tv.cards = nil
      @items = Tv.CardsAsItems
      DUMP(@items)

      # Tv::cards == $[]
      Tv.cards = []
      @items = Tv.CardsAsItems
      DUMP(@items)

      # Tv::cards == card1
      Tv.cards = [
        {
          "module"        => "bttv",
          "parameters"    => { "param1" => "testvalue" },
          "unique_key"    => "hluk.HvsIesmXXX1",
          "sound_card_no" => 0
        }
      ]
      @items = Tv.CardsAsItems
      DUMP(@items)

      # Tv::cards == card1, nil, nil, card4
      Tv.cards = [
        {
          "module"        => "bttv",
          "parameters"    => { "param1" => "testvalue" },
          "unique_key"    => "hluk.HvsIesmXXX1",
          "sound_card_no" => 0
        },
        nil,
        nil,
        {
          "module"        => "xyz",
          "parameters"    => { "par_other" => "testvalue2" },
          "name"          => "My TV_card",
          "unique_key"    => "j7F3.J8dhgs34fK2",
          "sound_card_no" => 1
        }
      ]
      @items = Tv.CardsAsItems
      DUMP(@items)

      nil
    end
  end
end

Yast::TvCardsAsItemsClient.new.main
