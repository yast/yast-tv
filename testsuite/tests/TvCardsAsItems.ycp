{

    include "testsuite.ycp";

    map READ = $[
	"target" : $[ "size": -1, "tmpdir" : "/tmp", ]
    ];

    TESTSUITE_INIT([READ], nil);
    import "Tv";

	// Tv::cards == nil
	Tv::cards = nil;
	list items = Tv::CardsAsItems();
	DUMP(items);

	// Tv::cards == $[]
	Tv::cards = [];
	items = Tv::CardsAsItems();
	DUMP(items);

	// Tv::cards == card1
	Tv::cards =
	    [ $[ "module" : "bttv",
		 "parameters" : $[ "param1" : "testvalue" ],
		 "unique_key" : "hluk.HvsIesmXXX1",
		 "sound_card_no" : 0
	      ]
	    ];
	items = Tv::CardsAsItems();
	DUMP(items);

	// Tv::cards == card1, nil, nil, card4
	Tv::cards =
	    [ $[ "module" : "bttv",
		 "parameters" : $[ "param1" : "testvalue" ],
		 "unique_key" : "hluk.HvsIesmXXX1",
		 "sound_card_no" : 0
	      ],
	      nil,
	      nil,
	      $[ "module" : "xyz",
		 "parameters" : $[ "par_other" : "testvalue2" ],
		 "name" : "My TV_card",
		 "unique_key" : "j7F3.J8dhgs34fK2",
		 "sound_card_no" : 1
	      ]
	    ];
	items = Tv::CardsAsItems();
	DUMP(items);
}
