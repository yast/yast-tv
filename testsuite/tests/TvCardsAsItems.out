Read	.target.ycp ["/var/lib/YaST2/installmap.ycp", $[]] $[]
Read	.probe.architecture nil
Read	.probe.has_apm nil
Read	.probe.has_pcmcia nil
Read	.probe.has_smp nil
Read	.probe.system []
Dump	[]
Dump	[]
Dump	[`item (`id (0), "0", "")]
Dump	[`item (`id (0), "0", ""), `item (`id (3), "3", "My TV_card")]
